lib/ansible/plugins/action/reboot.py

```diff
@@ -96,7 +106,13 @@ def construct_command(self):
     def get_system_boot_time(self):
         stdout = b''
         stderr = b''
-        command_result = self._low_level_execute_command(self.DEFAULT_BOOT_TIME_COMMAND, sudoable=self.DEFAULT_SUDOABLE)
+
+        # Determine the system distribution in order to use the correct shutdown command arguments
+        uname_result = self._low_level_execute_command('uname')
```

Comment:
I would have used platform.system rather than uname, but I see that's also what is used in the rest of the code.

Reply:
I agree. But in order to do that it would end up being `self._low_level_execute_command((python -c 'import platform; print(platform.system())')` since the action plugin is running on the control node, not the managed node. And that would also need to take into account `ansible_python_interpreter` on the remote node, further complicating matters.

For 2.8, I want to add proper platform support since how it is today isn't scaling well. The commands and parameters between various *nix distributions varies a lot more than I had hope.

I have a (very rough) WIP PR #45656 where I welcome your feedback.

Reply:
Platform is also not present in Python 3.8 i think

Reply:
sivel pointed out that platform.system() is not one of the things deprecated in the platform stdlib library.  platform.dist() is the main thing that's going away.